cs4360/apps/crm/api/src/app/dao/user/user.controller.ts

```diff
@@ -1,24 +1,45 @@
-import { Controller } from '@nestjs/common';
-import {Crud} from "@nestjsx/crud";
+import {Controller, Type} from '@nestjs/common';
+import {Crud, CrudController, CrudOptions, CrudRequest, Override, ParsedBody, ParsedRequest} from "@nestjsx/crud";
 import {UserEntity} from "@crm/nest/entities";
 import {UserService} from "./user.service";
+import {CreateUserRequest, CreateUserResponse} from "@crm/shared";
 
 
 @Crud({
   model: {
     type: UserEntity
   },
   params:{
-    id:{
+    id: {
       field: 'id',
       type: 'uuid',
       primary: true
     }
-  }
+  },
+  query: {
+    join: {
+      userType: {
+        eager: true, // deep copy down
+      },
+      contact: {
+        eager: true,
+        allow: ['id', 'firstName', 'lastName']
+      }
+    },
+    exclude: ['passwordHash']
+  },
 })
+@Controller('users')
+export class UserController implements CrudController<UserEntity> {
+  constructor(public service: UserService) {}
 
+  get base(): CrudController<UserEntity> {
+    return this;
+  }
 
-@Controller('user')
-export class UserController {
-  constructor(public service: UserService) {}
+  @Override()
+  async createOne(@ParsedRequest() req: CrudRequest, @ParsedBody() dto: CreateUserRequest): Promise<CreateUserResponse> {
+    const user: UserEntity = await this.base.createOneBase(req, dto);
+    return { id: user.id };
+  }
```

Comment:
This is how we change a typical response from NestJSX/CRUD response

https://github.com/nestjsx/crud/wiki/Controllers#routes-override

Reply:
good work